exasol_advanced_analytics_framework/udf_communication/peer_communicator/register_peer_sender.py

```diff
@@ -56,6 +56,9 @@ def _should_we_send(self):
         result = is_time and not self._finished
         return result
 
+    def is_ready_to_stop(self):
+        return (self._finished and self._needs_to_send_for_peer) or not self._needs_to_send_for_peer
```

Comment:
same as above, can be simplified to:
```suggestion
        return self._finished or not self._needs_to_send_for_peer
```